### DISCUSSION TITLE
Possible to use resource discovery across multiple subscriptions with separate dedicated scrapers?

### DISCUSSION THREAD

**Discussion by dks0296586** [tomkerkhove/promitor#2135] [Q&A]
Hello,

We have a large azure footprint that will eventually span multiple subscriptions.
I am running Promitor Resource Discovery and Scrapers in kubernetes and deploying them with Flux and kustomize so I can reuse much of the configurations.

I am testing out the possibility of using a single resource discovery agent that has access to multiple subscriptions, which seems to work as expected. 

I had then setup a scraper for subA(which has no resources yet).  subB has a few functionapps, postgres, etc.. resources.  What I saw was that the scraper for subA was now pulling metrics for subB resources due to the resource discovery groups including resources from both subscriptions.

Is there anyway for the scraper in subA to only collect metrics for subA resources and ignore subB resources?

I believe I incorrectly thought that azureMetadata.subscriptionId would work to target only that specified subscription.

I know it would be possible to create subscription specific resource groups in Resource discovery then set my metric declaration to use those groups, but that seems like that would cause a lot of extra configuration management between both the resource discovery and scraper agents.

**Comment by tomkerkhove** [comment]
First of all, sorry for the slow response!

> Is there anyway for the scraper in subA to only collect metrics for subA resources and ignore subB resources?
> 
> I believe I incorrectly thought that azureMetadata.subscriptionId would work to target only that specified subscription.

The only way to achieve this today is by using different discovery groups, unfortunately.

The subscription ID is only used for metric declaration which means for the metrics-as-code scenarios (without discovery).
https://docs.promitor.io/unreleased/scraping/overview/#general-declaration

> I know it would be possible to create subscription specific resource groups in Resource discovery then set my metric declaration to use those groups, but that seems like that would cause a lot of extra configuration management between both the resource discovery and scraper agents.

That's the current situation indeed, but let's think long-term - Would it help if you could apply basic filters on the scraper side? Think only `subscriptionId` and optionally `resourceGroup` (one or more)?

**Reply by dks0296586** [reply]
> That's the current situation indeed, but let's think long-term - Would it help if you could apply basic filters on the scraper side? Think only `subscriptionId` and optionally `resourceGroup` (one or more)?

That's exactly what I had in mind. Probably at the metric declaration level adding subscriptionId and resourceGroup and either filtering the results from inside the scraper, or enhancing the resource discovery api to add the additional parameters and filter the response there.

I haven't looked to in-depth at the code yet, so I'm not sure what the performance impact might be on the scraper to need to filter through  the resources from many subscriptions which is why I was initially thinking doing the filter in resource discovery might be beneficial.

**Reply by tomkerkhove** [reply]
Would you mind opening a feature request for this please? I think, however, that the resource discovery should do the filtering and not the scraper to reduce the amount of network calls.
